GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -225,16 +225,42 @@ protected override void Dispose(bool disposing)
         {
             if (disposing)
             {
+                //// MenuCommands not disposable
+                //// _superprojectCurrentCheckout not disposable
+                //// _selectedObjectIds not disposable
+                //// _baseCommitToCompare not disposable
                 _revisionSubscription?.Dispose();
                 _revisionReader?.Dispose();
+                //// _artificialStatus not disposable
+                //// _ambiguousRefs not disposable
+                //// _refFilterOptions not disposable
+                _lastVisibleResizableColumn?.Dispose();
+                _maximizedColumn?.Dispose();
+                //// _revisionGraphColumnProvider not disposable
+                _selectionTimer?.Dispose();
                 _buildServerWatcher?.Dispose();
 
                 if (_indexWatcher.IsValueCreated)
                 {
                     _indexWatcher.Value.Dispose();
                 }
 
+                //// _authorHighlighting not disposable
+                //// _parentChildNavigationHistory not disposable
+                //// _quickSearchProvider not disposable
+                //// _toolTipProvider  not disposable
+                _loadingControlSync.Dispose();
+                _loadingControlAsync.Dispose();
+                //// _navigationHistory not disposable
+                _revisionFilter?.Dispose();
```

Comment:
💡 `_revisionFilter` is never null here.

```suggestion
                _revisionFilter.Dispose();
```

Reply:
https://github.com/gitextensions/gitextensions/pull/7394#discussion_r361853159 and https://github.com/gitextensions/gitextensions/pull/7394#discussion_r362062838:
Although the ctor or `Setup` have created some object, it could have been destroyed by some method.
TearDown and Dispose should not make assumptions on the lifetime of member objects.
They should cleanup as much as possible. Afterwards they can explicitely assert non-null objects.

Reply:
This is not a test class. The linked comments do not apply to this case.

Reply:
Although the ctor has created some object, it could have been destroyed by some method.
`Dispose` should not make assumptions on the lifetime of member objects.
`Dispose` should cleanup as much as possible. _Afterwards_, it can explicitely assert member objects which shall not be `null` according to the design of the class.

(`Setup` and `TearDown` are analogous to ctor/`Dispose`. Hence the same reasoning applies to tests, too.)

Reply:
You're right -- for a different reason: This field cannot be `null` because it is `readonly`.

Reply:
C# does not behave like C++ in this regard. `Dispose(true)` will only be called if the constructor completes without throwing an exception, so anything assigned non-null in the constructor and not assigned elsewhere will definitely be non-null in `Dispose(bool)` if `disposing` is true. We check this parameter a few lines above the one I commented on.